src/builtins/cd.rs

```diff
@@ -124,12 +114,6 @@ pub fn cd(parser: &Parser, streams: &mut IoStreams, args: &mut [&wstr]) -> Optio
             }
         };
 
-        // We need to keep around the fd for this directory, in the parser.
-        let dir_fd = Arc::new(fd);
-
-        // Stash the fd for the cwd in the parser.
-        parser.libdata_mut().cwd_fd = Some(dir_fd);
```

Comment:
Since this was explicitly re-added to the parser fairly recently (for possible concurrent execution) I'd keep the `cwd_fd`.

I'm not sure why a directory would ever have `--x` permissions (why keep the execute bit, who are you helping?) but I guess we could and should support that case.
chdir() first and open() later sounds fine to me.

Reply:
I would close this in favor of #10436